### PR TITLE
DCGEO-209: increase cmsid column size

### PR DESCRIPTION
**Pull request by ericbuckley**
### Fixes [DCGEO-209](https://jira.cms.gov/browse/DCGEO-209)

The addition of the [TEST ACO Model](https://github.com/CMSgov/bcda-app/pull/726) along with its new format, `TEST\d{3}`, means some CMS IDs will be longer than 5 characters.  This PR updates the column length so we can store testing CMS IDs.

### Proposed Changes

- change the length of the cms id from 5 to 8 in `acos`, `cclf_files` & `suppressions` tables

### Change Details

- a new up/down migration for the change
- updating the `db/api.sql` fixture with the correct column types

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [ ] no PHI/PII is affected by this change

### Feedback Requested

- The migration could take some time if any of the above tables have a large number of rows.  Can someone confirm that the number of rows in production for the above tables is in the "hundreds or thousands" and not "millions"?

### CODE REVIEW

File: bcda/models/postgres/repository_test.go [hunk 1]
```diff
@@ -482,7 +482,7 @@ func (r *RepositoryTestSuite) TestACOMethods() {
 		r.repository.UpdateACO(ctx, uuid.Parse(aco.ClientID),
 			map[string]interface{}{"termination_details": &models.Termination{}}),
 		fmt.Sprintf("ACO %s not updated, no row found", aco.ClientID))
-	assert.Contains(r.repository.CreateACO(ctx, aco).Error(), "duplicate key value violates unique constraint \"acos_cms_id_key\"")
+	assert.Contains(r.repository.CreateACO(ctx, aco).Error(), "duplicate key value violates unique constraint \"acos_pkey\"")
```
Comment (ericbuckley) [inline bcda/models/postgres/repository_test.go:485]: NOTE: the ACOS table has 2 constraints; a unique primary key and unique cmsid.
the initial migration, 000001, created the former constraint before the latter,
thus violations with both will report the former error.  However with
migration 000015, altering the column deletes, then recreates the
unique cmsid, thus placing it second.  This test case, that violates
both constraints, is now going to report an issue with the primary key
as that is first in the list of table constraints

**Comment by brianburton** [comment]
Couple of questions:

- Why are there two migrations that affect the same tables/columns?
- How do you know the two migrations will run in a specific order?  Are they triggered directly by a script?

**Comment by ericbuckley** [comment]
> Couple of questions:
> 
>     * Why are there two migrations that affect the same tables/columns?
> 
>     * How do you know the two migrations will run in a specific order?  Are they triggered directly by a script?

One is an up migration (for going forwards) and the other is a down (for going backwards).  https://github.com/golang-migrate/migrate/blob/master/MIGRATIONS.md

**Review by brianburton — APPROVED** [review]
Looks good!

**Review by brianburton — APPROVED** [review]
Looks good to me.

**Review by KyRobbins — APPROVED** [review]

**Comment by Gabasol** [comment]
@ericbuckley - can you kindly add the BCDA devs to this PR to review? Thanks!

File: db/migrations/bcda/000015_increase_cmsid_size.up.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-- Increasing the size of CMSID columns from 5 to 8
+
+BEGIN;
+
+ALTER table public.acos ALTER COLUMN cms_id type character varying(8);
+ALTER table public.cclf_files ALTER COLUMN aco_cms_id type character varying(8);
+ALTER table public.suppressions ALTER COLUMN aco_cms_id type character varying(8);
+
+COMMIT;
```
[the remark this answers is not in the document]
Comment (ericbuckley) [reply db/migrations/bcda/000015_increase_cmsid_size.up.sql:9]: I manually updated that file to account for the 3 varchar changes in this migration

**Review by CosmicSyntax — APPROVED** [review]
🍺

**Review by SMLuthi — APPROVED** [review]
LGTM!

**Pull request merged**